### PR TITLE
Add initial cookbooks and fill-in additional details on processors

### PR DESCRIPTION
**Pull request by ryanjdew**

### CODE REVIEW

File: _data/sidenav.yml [hunk 1]
```diff
@@ -3,5 +3,11 @@ main:
   - { url: '/getting-started', text: 'Getting Started' }
   - { url: '/step-by-step', text: 'Step-by-Step Guide' }
   - { url: '/nifi-features', text: 'NiFi Features' }
-  - { url: '/cookbook-recipes', text: 'Cookbook Recipes' }
+  - { url: '/cookbook-recipes', text: 'Cookbook Recipes',
+      childPages: [
+        { url: '/get-data-from-a-relational-database', text: 'Relational Database' },
```
Comment (dmcassel) [inline _data/sidenav.yml]: I think a consistent naming pattern would be useful

```suggestion
        { url: '/get-data-from-a-relational-database', text: 'Relational Database to MarkLogic' },
```

File: _pages/cookbooks/cookbook-recipes.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+title: Cookbook Recipes
+lead_text: ''
+permalink: /cookbook-recipes
+---
+
+# Cookbook Recipes
+
+## [Relational Database][relational-database]
+
+Pull data into MarkLogic from a relational database source as JSON documents.
```
Comment (dmcassel) [inline _pages/cookbooks/cookbook-recipes.md]: Phrasing is a little awkward. Rephrase:

```suggestion
Pull data from a relational database source and write into MarkLogic as JSON documents.
```

File: _pages/cookbooks/cookbook-recipes.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+## [File System][file-system]
+
+Pull data into MarkLogic from a file system directory.
+
+[Read More][file-system]
+
+## [Data Hub Input Flow][dhf-input-flow]
+
+Use the `Server Transform` property and dynamic `trans:*` properties to run a Data Hub Input Flow transform against the input documents.
```
Comment (dmcassel) [inline _pages/cookbooks/cookbook-recipes.md]: Skip the implementation details at this level. 

```suggestion
Use a [MarkLogic Data Hub](https://github.com/marklogic/marklogic-data-hub) Input Flow to transform documents while loading.
```

File: _pages/cookbooks/file-system-to-marklogic.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
+---
+layout: inner
+title: File System to MarkLogic
+lead_text: ''
+permalink: /file-system-to-marklogic
+---
+
+# File System to MarkLogic
+
+This example watches a directory for files, imports them into MarkLogic, then deletes them. The MarkLogic URI is /files/ followed by the filename. The NiFi filename property is set by the GetFile processor.
```
Comment (dmcassel) [inline _pages/cookbooks/file-system-to-marklogic.md]:
```suggestion
This example watches a directory for files, imports them into MarkLogic, then deletes them from the file system. The MarkLogic URI is `/files/` followed by the filename. The NiFi filename property is set by the `GetFile` processor. 
```

File: _pages/cookbooks/file-system-to-marklogic.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+Many NiFi properties support embedded expressions, such as ${filename} in the ml.uri property below. For more about expressions, see [Apache NiFi Expression Language Guide.][nifi-exp-lang]
+
+
+<a href="./files/FileSystemToMarkLogic.xml" download>Download Template</a>
+
+## Processors:
+
+### GetFile
+
+reads files from a watched directory
```
Comment (dmcassel) [inline _pages/cookbooks/file-system-to-marklogic.md]:
```suggestion
Reads files from a watched directory.
```

File: _pages/cookbooks/file-system-to-marklogic.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+**Properties**
+
+Input Directory
+ : */some/path*
```
Comment (dmcassel) [inline _pages/cookbooks/file-system-to-marklogic.md]: Any fancy formatting required? For instance, on Windows, can you do stuff like "C:\Users\dcassel", or does the path need to be altered at all? (Probably a question for Frank.)

File: _pages/cookbooks/get-data-from-relational.md [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+### Dataset
+
+Our relational cookbook uses the sample employee database located here:
+[https://github.com/datacharmer/test_db](https://github.com/datacharmer/test_db)
+
+Follow the instructions at the linked dataset repository to load your data.
+
+There are about 300,000 employees in the database, so we will assume that that is too big for ExecuteSQL without paging.
+
+Run the following SQL query which will create the `employee_detail` view we will use for extracting data.
```
Comment (dmcassel) [inline _pages/cookbooks/get-data-from-relational.md]:
```suggestion
Run the following SQL query, which will create the `employee_detail` view we will use for extracting data.
```

File: _pages/cookbooks/run-data-hub-input-flow.md [hunk 8]
```diff
@@ -0,0 +1,161 @@
+---
+layout: inner
+title: Run Data Hub Input Flow
+lead_text: ''
+permalink: /run-data-hub-input-flow
+---
+
+# Run Data Hub Input Flow
+
+This template demonstrates how to ingest a document and transform with a Data Hub Framework input flow.
```
Comment (dmcassel) [inline _pages/cookbooks/run-data-hub-input-flow.md]:
```suggestion
This template demonstrates how to ingest a document and transform it with a Data Hub Framework input flow.
```

File: _pages/cookbooks/file-system-to-marklogic.md (hunk 6 above)
Comment (ryanjdew) [reply _pages/cookbooks/file-system-to-marklogic.md]: You don't need to do any special formatting for the path from what I've seen. The original templates had a Windows path and they followed the standard format like "C:\Users\dcassel".

**Pull request merged**